sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala

```diff
@@ -171,6 +238,19 @@ case class CustomShuffleReaderExec private(
         } else {
           Map.empty
         }
+      } ++ {
+        if (isLocalReader) {
+          Map.empty
+        } else {
+          if (hasCoalescedPartition) {
+            Map("numCoalescedPartitions" ->
+              SQLMetrics.createMetric(sparkContext, "number of coalesced partitions"),
+              "numPartitionsToCoalesce" ->
```

Comment:
There is always a shuffle node below `CustomShuffleReader`. I think it makes more sense to let the shuffle node to report the metrics of the number of partitions/reducers.

Reply:
I added the coalesce metrics in the same way skew-related metrics were done.
Are you suggesting that all of them move to the `ShuffleExchangeExec` below?